File: packages/dll/src/net/messages/getstatic.cpp
#include "getstatic.h"
#include "../message.h"
#include "../../injector.h"
#include "../../UE/v425/uobjectarray.h"

GetStaticRequest::GetStaticRequest(Buffer& data) : Request(data), name(data.Read()) {}

void GetStaticRequest::Process() {
	GetStaticResponse res;

	WFUObjectItem item = injector.objectArray.FindObject(name);

	if (item) {
		WFUObjectItem staticInstance = injector.objectArray.FindStatic(item.GetObject());
		if (staticInstance) {
			res.name = staticInstance.GetObject().GetFullName();
		}
	}

	ProcessResponse(res);
}

void GetStaticResponse::Process(Buffer& data) {
	Response::Process(data);

	data.Write(name);
}

File: packages/electron/src/main/net/messages/getStatic.ts
import { RROxApp } from "../../app";
import { BufferIO } from "../io";
import { MessageType, Request, Response } from "../message";

export class GetStaticRequest extends Request {
    public name: string;

    constructor( app: RROxApp, name: string ) {
        super( app, MessageType.GET_STATIC );

        this.name = name;
    }

    public process( data: BufferIO ) {
        super.process( data );

        data.writeString( this.name );
    }
}

export class GetStaticResponse extends Response {
    public name?: string;

    constructor( app: RROxApp, data: BufferIO ) {
        super( app, data );
        
        this.name = data.readString();
    }
}